packages/explat-client-react-helpers/src/index.tsx

```diff
@@ -25,25 +25,35 @@ interface ExPlatClientReactHelpers {
 		name: string;
 		children: { default: React.ReactNode; treatment: React.ReactNode; loading: React.ReactNode };
 	} ) => JSX.Element;
+
+	/**
+	 * Inject experiment data into a child component.
+	 * Use when hooks aren't available.
+	 */
+	ProvideExperimentData: ( props: {
+		children: (
+			isLoading: boolean,
+			experimentAssignment: ExperimentAssignment | null
+		) => JSX.Element;
+		name: string;
+	} ) => JSX.Element;
 }
 
 export default function createExPlatClientReactHelpers(
 	exPlatClient: ExPlatClient
 ): ExPlatClientReactHelpers {
 	const useExperiment = ( experimentName: string ): [ boolean, ExperimentAssignment | null ] => {
 		const [ previousExperimentName ] = useState( experimentName );
-		const [ isLoading, setIsLoading ] = useState< boolean >( true );
-		const [
-			experimentAssignment,
-			setExperimentAssignment,
-		] = useState< ExperimentAssignment | null >( null );
+		const [ state, setState ] = useState< [ boolean, ExperimentAssignment | null ] >( [
+			true,
+			null,
+		] );
```

Comment:
This makes me wish for C# named tuples...

```csharp
var tuple = ( isLoading: false, assignment: null );
// use as tuple.isLoading or tuple.Item0
```

Reply:
I could use a map actually... but I might leave it like this for now.

Reply:
Yeah, I think it is fine. I think an array is a more idiomatic way of doing it. Sorry, I wasn't saying that it needed a change; just lamenting the lack of tuples.

Reply:
I believe you could also keep using two separate atomic `useState`s for `isLoading` and `experimentAssignment`.

Even though they're set with two distinct `setState` calls, it should be guaranteed by React that the values are always in sync. I.e., render is never done with only one applied.

Reply:
> I believe you could also keep using two separate atomic useStates for isLoading and experimentAssignment.
> 
> Even though they're set with two distinct setState calls, it should be guaranteed by React that the values are always in sync. I.e., render is never done with only one applied.

Yeah that is what I thought, [but in tests I found that two renders occurred](https://github.com/Automattic/wp-calypso/pull/50993/files/684f47daccb6b2300ecaaad2584f71fc41f7244d#diff-910569eaf3e8b5a6129b7c2f3102c9569aff72f81f782584981cc5bffe5bd487R177).

Reply:
> Yeah that is what I thought, but in tests I found that two renders occurred.

You're right! Even in a real app, it depends on where exactly the `setState` calls are done. In an event handler, state updates are batched, elsewhere they aren't.

This will do only one render:
```js
const inc = x => x + 1;
<button onClick={ () => { setA( inc ); setB( inc ); } } />
```
This will do two renders because we're outside a DOM event handler:
```js
<button onClick={ () => setTimeout( () => { setA( inc ); setB( inc ); }, 0 ) } />

Reply:
It is quite a catch that it works so specifically in event handlers, it shows that Async + React Hooks can be quite subtle.

Reply:
TIL